File: AINovalServer/src/main/java/com/ainovel/server/web/dto/ImportStatus.java
package com.ainovel.server.web.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * 小说导入状态DTO
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ImportStatus {

    /**
     * 状态码：PROCESSING, SAVING, INDEXING, COMPLETED, FAILED, ERROR, CANCELLED
     */
    private String status;

    /**
     * 状态详细信息
     */
    private String message;

    /**
     * 进度百分比（可选）
     */
    private Double progress;

    /**
     * 当前步骤名称
     */
    private String currentStep;

    /**
     * 详细步骤列表
     */
    private List<ImportStep> steps;

    /**
     * 估算剩余时间（秒）
     */
    private Integer estimatedRemainingSeconds;

    /**
     * 处理的章节数量
     */
    private Integer processedChapters;

    /**
     * 总章节数量
     */
    private Integer totalChapters;

    /**
     * 已生成摘要的章节数量
     */
    private Integer summarizedChapters;

    /**
     * 错误列表
     */
    private List<String> errors;

    /**
     * 警告列表
     */
    private List<String> warnings;

    // 保持向后兼容的构造函数
    public ImportStatus(String status, String message) {
        this.status = status;
        this.message = message;
        this.progress = null;
    }

    public ImportStatus(String status, String message, Double progress) {
        this.status = status;
        this.message = message;
        this.progress = progress;
    }
}

File: AINoval/lib/models/import_status.dart
/// 小说导入状态模型
class ImportStatus {
  /// 从JSON创建实例
  factory ImportStatus.fromJson(Map<String, dynamic> json) {
    return ImportStatus(
      status: json['status'] as String,
      message: json['message'] as String,
      progress: (json['progress'] as num?)?.toDouble(),
      currentStep: json['currentStep'] as String?,
      processedChapters: json['processedChapters'] as int?,
      totalChapters: json['totalChapters'] as int?,
    );
  }

  /// 创建导入状态
  ImportStatus({
    required this.status,
    required this.message,
    this.progress,
    this.currentStep,
    this.processedChapters,
    this.totalChapters,
  });

  /// 导入状态 (PROCESSING, SAVING, INDEXING, COMPLETED, FAILED, ERROR)
  final String status;

  /// 状态消息
  final String message;

  /// 导入进度 (0.0 - 1.0)
  final double? progress;

  /// 当前步骤描述
  final String? currentStep;

  /// 已处理章节数
  final int? processedChapters;

  /// 总章节数
  final int? totalChapters;

  /// 转换为JSON
  Map<String, dynamic> toJson() {
    return {
      'status': status,
      'message': message,
      if (progress != null) 'progress': progress,
      if (currentStep != null) 'currentStep': currentStep,
      if (processedChapters != null) 'processedChapters': processedChapters,
      if (totalChapters != null) 'totalChapters': totalChapters,
    };
  }

  @override
  String toString() => 'ImportStatus{status: $status, message: $message, progress: $progress, currentStep: $currentStep, processedChapters: $processedChapters, totalChapters: $totalChapters}';
}
